spec/core/terraform/config/dsl_spec.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require "spec_helper"
+
+describe Terraform::Config::Dsl do
+  include described_class
+
+  context "with simple config" do
+    subject(:generated) do
```

Comment:
Same here and for all specs below. I feel like assigning the value we want to test directly in the `it` block makes it easier to understand what's happening.

What are the benefits of using `subject`?

Reply:
1. I find it more readable when `subject` is defined in top level so we can see what is being tested right away
2. Other `it` blocks can be added more easily.

If we move `subject` to `it` blocks code will look like this:
```
context "with simple config" do # Now we have context which does nothing other than adding some description
  it "generates correct config" do
    generated =
      block :block_name do # rubocop:disable Lint/EmptyBlock
      end

    expect(generated).to eq(
      <<~EXPECTED
        block_name {
        }
      EXPECTED
    )
  end
end
```
Or like this:
```
it "generates correct output for simple config" do # Now we have context description in `it` block
  generated =
    block :block_name do # rubocop:disable Lint/EmptyBlock
    end

  expect(generated).to eq(
    <<~EXPECTED
      block_name {
      }
    EXPECTED
  )
end
```

Reply:
I personally find it more readable to define the subject inside the `it` block, because we concentrate what's needed for the test inside of the block (excluding potential initializations and `before` / `after` blocks). Plus, if we add other `it` blocks we might not use the same subject, which could make things confusing.

But I won't push on that here. And since the `context`s are well defined, we'd likely use the same subject if we added more `it` blocks to these specs, so ok.